jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -953,6 +1177,18 @@ def _curve_of_growth(data, centroid, aperture, final_sum, wcs=None, background=0
     """
     n_datapoints += 1  # n + 1
 
+    # determined desired unit for output sum array and y label
+    # cubeviz only to handle unit conversion display unit changes
+    if display_unit is not None:
+        sum_unit = u.Unit(display_unit)
+    else:
+        if isinstance(data, u.Quantity):
+            sum_unit = data.unit
+        else:
+            sum_unit = None
+    if sum_unit and pixarea_fac is not None:
+        sum_unit *= pixarea_fac.unit

```

Comment:
I don't understand what this is doing. As discussed, in `main`, JWST data with MJy/sr naturally have the `/sr` removed during curve of growth calculation. Theoretically, this function should not have to change at all in this PR. Why is this change needed again?

Reply:
the initial sum_unit is always the data unit (MJy / sr), and if it is multiplied by pixarea_fac (which is how it is removed for the final sum / curve of growth) it will be MJy. This block handles translating the output sum units to the display unit if they differ